src/lib/util/Cache/MemberTags.ts

```diff
@@ -48,16 +54,22 @@ export class MemberTags extends Collection<string, MemberTag> {
 	public async fetch(): Promise<this>;
 	public async fetch(id?: string): Promise<MemberTag | null | this> {
 		if (typeof id === 'undefined') {
-			const members = await this.guild.members.fetch();
-			for (const member of members.values()) this.create(member);
+			if (this.kFetchAllPromise === null) {
+				this.kFetchAllPromise = this.guild.members.fetch().then(members => {
+					for (const member of members.values()) this.create(member);
+					this.kFetchAllPromise = null;
```

Comment:
This should be moved to a `finally` function in case the promise throws, as to prevent an ever hanging promise.

Reply:
Resolved in f011b56.